rero_ils/modules/locations/api.py

```diff
@@ -189,3 +189,40 @@ def bulk_index(self, record_id_iterator):
         :param record_id_iterator: Iterator yielding record UUIDs.
         """
         super().bulk_index(record_id_iterator, doc_type='loc')
+
+
+def search_locations_by_pid(organisation_pid=None, library_pid=None,
+                            is_online=False, is_pickup=False,
+                            sort_by_field='location_name', sort_order='asc',
+                            preserve_order=False):
+    """Retrieve locations attached to the given organisation or library."""
+    search = LocationsSearch()
+
+    if organisation_pid:
+        search = search.filter('term', organisation__pid=organisation_pid)
+    elif library_pid:
+        search = search.filter('term', library__pid=library_pid)
+    else:
+        raise MissingRequiredParameterError(
+            "One of the parameters 'organisation_pid' "
+            "or 'library_pid' is required."
+        )
+
+    if is_online:
+        search = search.filter('term', is_online=is_online)
+
+    if is_pickup:
+        search = search.filter('term', is_pickup=is_pickup)
+
+    if sort_by_field:
+        search = search.sort({sort_by_field: {'order': sort_order}})
+        if preserve_order:
+            search = search.params(preserve_order=True)
+    return search
+
+
+def search_location_by_pid(loc_pid):
+    """Search location for the given location pid."""
+    loc_search = LocationsSearch().filter('term', pid=loc_pid)
+    for location in loc_search.scan():
+        return location
```

Comment:
Did you try this method with a dummy `loc_pid` argument ?
In other implementation, the idiom `try... except StopIteration` is used.

Reply:
I agree, this should be handled if no location is found.

Reply:
This is not necessary. scan() return a generator that will iterate over all the documents matching the query otherwise it returns an empty generator.
Tested with dummy loc_pid and it works as expected.